src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -16324,125 +16460,408 @@ static void Main()
-                    IL_0056:  ldarg.0
-                    IL_0057:  ldloc.3
-                    IL_0058:  stfld      "System.Runtime.CompilerServices.TaskAwaiter<T> Program.<F>d__2<T>.<>u__1"
-                    IL_005d:  ldarg.0
-                    IL_005e:  ldflda     "System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Collections.Generic.List<T>> Program.<F>d__2<T>.<>t__builder"
-                    IL_0063:  ldloca.s   V_3
-                    IL_0065:  ldarg.0
-                    IL_0066:  call       "void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Collections.Generic.List<T>>.AwaitUnsafeOnCompleted<System.Runtime.CompilerServices.TaskAwaiter<T>, Program.<F>d__2<T>>(ref System.Runtime.CompilerServices.TaskAwaiter<T>, ref Program.<F>d__2<T>)"
-                    IL_006b:  leave.s    IL_00e4
-                    IL_006d:  ldarg.0
-                    IL_006e:  ldfld      "System.Runtime.CompilerServices.TaskAwaiter<T> Program.<F>d__2<T>.<>u__1"
-                    IL_0073:  stloc.3
-                    IL_0074:  ldarg.0
-                    IL_0075:  ldflda     "System.Runtime.CompilerServices.TaskAwaiter<T> Program.<F>d__2<T>.<>u__1"
-                    IL_007a:  initobj    "System.Runtime.CompilerServices.TaskAwaiter<T>"
-                    IL_0080:  ldarg.0
-                    IL_0081:  ldc.i4.m1
-                    IL_0082:  dup
-                    IL_0083:  stloc.0
-                    IL_0084:  stfld      "int Program.<F>d__2<T>.<>1__state"
-                    IL_0089:  ldloca.s   V_3
-                    IL_008b:  call       "T System.Runtime.CompilerServices.TaskAwaiter<T>.GetResult()"
-                    IL_0090:  stloc.2
-                    IL_0091:  ldarg.0
-                    IL_0092:  ldfld      "System.Collections.Generic.List<T> Program.<F>d__2<T>.<>7__wrap1"
-                    IL_0097:  ldloc.2
-                    IL_0098:  callvirt   "void System.Collections.Generic.List<T>.Add(T)"
-                    IL_009d:  ldarg.0
-                    IL_009e:  ldfld      "System.Collections.Generic.List<T> Program.<F>d__2<T>.<>7__wrap2"
-                    IL_00a3:  ldarg.0
-                    IL_00a4:  ldfld      "T Program.<F>d__2<T>.z"
-                    IL_00a9:  callvirt   "void System.Collections.Generic.List<T>.Add(T)"
-                    IL_00ae:  ldarg.0
-                    IL_00af:  ldfld      "System.Collections.Generic.List<T> Program.<F>d__2<T>.<>7__wrap2"
-                    IL_00b4:  stloc.1
-                    IL_00b5:  leave.s    IL_00d0
+                    IL_000d:  br.s       IL_001d
+                    IL_000f:  ldloc.1
+                    IL_0010:  callvirt   "string System.Collections.Generic.IEnumerator<string>.Current.get"
+                    IL_0015:  stloc.2
+                    IL_0016:  ldloc.0
+                    IL_0017:  ldloc.2
+                    IL_0018:  callvirt   "void System.Collections.Generic.List<object>.Add(string)"
```

Comment:
uh..which one? :)

Reply:
This should be the specialized `List<object>.Add(string)` rather than `List<object>.Add(object)`.